sink.go

```diff
@@ -12,6 +12,9 @@ type MetricSink interface {
 	SetGauge(key []string, val float32)
 	SetGaugeWithLabels(key []string, val float32, labels []Label)
 
+	SetGaugeFloat64(key []string, val float64)
+	SetGaugeFloat64WithLabels(key []string, val float64, labels []Label)
+
```

Comment:
Adding methods to an existing public interface is technically a breaking change which would require a whole new major version of the library with new imports etc.

That's because any external implementations of this metrics sink will no longer compile after an upgrade until they also implement these new methods.

I think there are two options here:
 1. Make this a v2 change which for go mod requires moving all the code to a new /v2 subdir with associated go.mod changes etc.
 2. Add these to a _different_ optional interface like `Float64Sink` or something (I know, names are hard...) then implement that for any internal sinks you care about. In the place where we call this method on a sink, type check it to see if it implements this interface first and if not do something. The choice on what to do here is somewhat nuanced. The signatures have no error returns for now because metrics are meant to be "fire and forget" and shouldn't hold up application processing, so the choice is either `panic` and document that it is a developer error to allow a sink that doesn't support this new extended interface to be configured in an app that relies on the new methods you added... or you could just auto-downgrade to float32 which at least results in _some_ metric though if the application bothered to use the 64 bit variant it's still likely to be the wrong thing, or you could just drop the gauge on the floor and maybe log it (though that might require plumbing a logger into the library) and if you do choose this path you probably would want to consider rate-limiting the logs because SetGauge could be called really often in an application and would fill logs really fast if every call logs a warning.

What do you think about those options?

Reply:
I prefer option 2. Do you have any preferences?

Reply:
2 seems more pragmatic to me. It's a little more "messy" in that ideally we'd have a single nice interface but IMO this isn't a big enough issue to warrant a v2 with all that entails.

The implicit option 3 would be to ask if this is even worth solving at all and choose to do nothing and live with the limitation of float32 for now. It's always worth asking the question! I can't say how important that is for vault though or what the implications of not fixing this might be so assuming it's worth it, I'd go for 2.

Reply:
I did some more thinking and I believe option 2 is also not possible, since it would break FanoutSink, which relies on every Sink implementing the same interface (MetricsSink).

Reply:
I think in order to fix the issue we would indeed need a v2 of the library. 😿